util/tracer/context.go

```diff
@@ -0,0 +1,38 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Tobias Schottdorf (tobias.schottdorf@gmail.com)
+
+package tracer
+
+import "golang.org/x/net/context"
+
+// contextKey is the key claimed for storing and retrieving a Trace from
+// a context.Context.
+const contextKey = "_tracer"
```

Comment:
The [examples](http://blog.golang.org/context) given by Google on how to do this use named integer types, like:

```
type key int

const contextKey key = 0
```

No idea if that would increase performance at all, but its also makes it impossible to collide with a key from anywhere else.